Creating more than 10 deployment variables fails

Hello

When creating more than 10 `bitbucket_deployment_variable` for the same `bitbucket_deployment`, any variable over the count of 10 in total throws an error, but are still created in bitbucket (which [have now increased the number of allowed variables from 10 to 50](https://community.atlassian.com/t5/Bitbucket-questions/Bitbucket-deployments-limit/qaq-p/1104607)).

Steps to reproduce:
* Create a deployment with 11 variables

Expected results:
* Deployment with 11 variables created and tracked in terraform state

Actual results:
* Deployment with 11 variables are created, but only the first 10 variables are tracked in terraform state

Debug output when creating a deployment with more than 10 variables:
```
...
2022-07-04T08:59:15.377+0200 [INFO]  provider.terraform-provider-bitbucket_v2.20.0: 2022/07/04 08:59:15 [DEBUG] Using API Basic Auth: timestamp=2022-07-04T08:59:15.377+0200
module.dev-vars.bitbucket_deployment_variable.terraform-config-path: Creating...
2022-07-04T08:59:15.378+0200 [DEBUG] module.dev-vars.bitbucket_deployment_variable.terraform-config-path: applying the planned Create change
2022-07-04T08:59:21.419+0200 [INFO]  provider.terraform-provider-bitbucket_v2.20.0: 2022/07/04 08:59:21 [WARN] Deployment Variable ({f7342582-cb2b-45d1-86f3-b8ed55a7c8d4}) not found, removing from state: timestamp=2022-07-04T08:59:21.419+0200
2022-07-04T08:59:21.420+0200 [ERROR] vertex "module.dev-vars.bitbucket_deployment_variable.terraform-config-path" error: Provider produced inconsistent result after apply
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to module.dev-vars.bitbucket_deployment_variable.terraform-config-path, provider
│ "provider[\"registry.terraform.io/drfaust92/bitbucket\"]" produced an unexpected new value: Root resource was present, but
│ now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

--- a/bitbucket/resource_deployment_variable_test.go
+++ b/bitbucket/resource_deployment_variable_test.go
@@ -54,7 +54,7 @@ func TestAccBitbucketDeploymentVariable_basic(t *testing.T) {
 func TestAccBitbucketDeploymentVariable_manyVars(t *testing.T) {
 	rName := acctest.RandomWithPrefix("tf-test")
 	owner := os.Getenv("BITBUCKET_TEAM")
-	resourceName := "bitbucket_deployment_variable.test[0]"
+	// resourceName := "bitbucket_deployment_variable.test[0]"
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
@@ -63,15 +63,6 @@ func TestAccBitbucketDeploymentVariable_manyVars(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config: testAccBitbucketDeploymentVariableManyConfig(owner, rName, "test", false),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckBitbucketDeploymentVariableExists(resourceName),
-				),
-			},
-			{
-				ResourceName:      resourceName,
-				ImportState:       true,
-				ImportStateIdFunc: testAccBitbucketDeploymentVariableImportStateIdFunc(resourceName),
-				ImportStateVerify: true,
 			},
 		},
 	})

--- a/bitbucket/resource_deployment_variable_test.go
+++ b/bitbucket/resource_deployment_variable_test.go
@@ -51,6 +51,32 @@ func TestAccBitbucketDeploymentVariable_basic(t *testing.T) {
 	})
 }
 
+func TestAccBitbucketDeploymentVariable_manyVars(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-test")
+	owner := os.Getenv("BITBUCKET_TEAM")
+	resourceName := "bitbucket_deployment_variable.test[0]"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckBitbucketDeploymentVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBitbucketDeploymentVariableManyConfig(owner, rName, "test", false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBitbucketDeploymentVariableExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateIdFunc: testAccBitbucketDeploymentVariableImportStateIdFunc(resourceName),
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
 func TestAccBitbucketDeploymentVariable_secure(t *testing.T) {
 	rName := acctest.RandomWithPrefix("tf-test")
 	owner := os.Getenv("BITBUCKET_TEAM")
@@ -156,6 +182,30 @@ resource "bitbucket_deployment" "test" {
 
 resource "bitbucket_deployment_variable" "test" {
   key        = "test"
+  value      = %[3]q
+  deployment = bitbucket_deployment.test.id
+  secured    = %[4]t
+}
+`, owner, rName, val, secure)
+}
+
+func testAccBitbucketDeploymentVariableManyConfig(owner, rName, val string, secure bool) string {
+	return fmt.Sprintf(`
+resource "bitbucket_repository" "test" {
+  owner = %[1]q
+  name  = %[2]q
+}
+
+resource "bitbucket_deployment" "test" {
+  name       = %[2]q
+  stage      = "Test"
+  repository = bitbucket_repository.test.id
+}
+
+resource "bitbucket_deployment_variable" "test" {
+  count = 50
+
+  key        = "test${count.index}"
   value      = %[3]q
   deployment = bitbucket_deployment.test.id
   secured    = %[4]t

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,7 @@
 module github.com/terraform-providers/terraform-provider-bitbucket
 
 require (
-	github.com/DrFaust92/bitbucket-go-client v0.3.0
+	github.com/DrFaust92/bitbucket-go-client v0.4.0
 	github.com/antihax/optional v1.0.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.24.1
 	github.com/satori/go.uuid v1.2.0

